File: server/app/schemas/notification.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, Field

class NotificationBase(BaseModel):
    title: str = Field(..., max_length=255, description="通知标题")
    content: str = Field(..., description="通知正文内容")
    notification_type: str = Field("system", description="通知类型")
    link_url: Optional[str] = Field(None, max_length=255, description="跳转链接")

class NotificationOut(NotificationBase):
    id: UUID
    user_id: UUID
    read_at: Optional[datetime] = None
    created_at: datetime

    class Config:
        from_attributes = True

File: web/src/api/modules/notification.ts
import request from '../request'

export interface NotificationOut {
  id: string
  title: string
  content: string
  notification_type: string
  link_url?: string
  user_id: string
  read_at?: string
  created_at: string
}

export const notificationApi = {
  listNotifications(params?: { unread_only?: boolean }) {
    return request.get('/notifications/', { params })
  },

  markAsRead(id: string) {
    return request.post(`/notifications/${id}/read`)
  },

  markAllAsRead() {
    return request.post('/notifications/read-all')
  }
}
